.circleci/config.yml

```diff
@@ -0,0 +1,170 @@
+        command: |
+          DOCKER_IMAGE_TAG="$(cksum ../docker/php-fpm/Dockerfile | cut -d ' ' -f1)"
+          yq write --inplace deployments/webserver.yml spec.template.spec.containers[0].image $DOCKER_USERNAME/$DOCKER_IMAGE:$DOCKER_IMAGE_TAG
+        working_directory: project-base/k8s
+
+    - run:
+        name: Create configmap manifests
+        command: |
+          kubectl create configmap nginx-configuration --from-file ../../docker/nginx/nginx.conf --dry-run --output=yaml > nginx-configuration.yml
+          kubectl create configmap php-configuration --from-file ../docker/php-fpm/php-ini-overrides.ini --dry-run --output=yaml > php-configuration.yml
+        working_directory: project-base/k8s
+
+    - run:
+        name: Setup domain a parameters files
+        command: |
+          FIRST_DOMAIN_HOSTNAME=$CIRCLE_BRANCH-circle.$DEVELOPMENT_SERVER_DOMAIN
+          SECOND_DOMAIN_HOSTNAME=2.$CIRCLE_BRANCH-circle.$DEVELOPMENT_SERVER_DOMAIN
+          # Copy files from teplates
+          cp project-base/app/config/domains_urls.yml.dist project-base/app/config/domains_urls.yml
+          cp project-base/app/config/parameters_test.yml.dist project-base/app/config/parameters_test.yml
+          cp project-base/app/config/parameters.yml.dist project-base/app/config/parameters.yml
+          # Add domains into domain file
+          yq write --inplace project-base/app/config/domains_urls.yml domains_urls[0].url http://$FIRST_DOMAIN_HOSTNAME
+          yq write --inplace project-base/app/config/domains_urls.yml domains_urls[1].url http://$SECOND_DOMAIN_HOSTNAME
+
+    - run:
+        name: Set SSH credentials
+        command: |
+          echo -e $SERVER_SSL_CERT > ~/shopsys
+          chmod 0400 ~/shopsys
+
+    - run:
+        name: Upload repository into cluster, create volumes and set correct permissions
+        command: |
+          cd .. && tar -zcvf source.tar.gz source
+          ssh -o "StrictHostKeyChecking no" -i ~/shopsys www-data@heimdall.shopsys.com mkdir -p /workspace/shopsys/$CIRCLE_BRANCH
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com rm -rf /workspace/shopsys/$CIRCLE_BRANCH/source
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com sudo chown -R www-data:www-data /workspace/shopsys/$CIRCLE_BRANCH/db
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com rm -rf /workspace/shopsys/$CIRCLE_BRANCH/db/*
+          scp -i ~/shopsys source.tar.gz www-data@heimdall.shopsys.com:/workspace/shopsys/$CIRCLE_BRANCH/source.tar.gz
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com tar -zxvf /workspace/shopsys/$CIRCLE_BRANCH/source.tar.gz -C /workspace/shopsys/$CIRCLE_BRANCH/
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com rm /workspace/shopsys/$CIRCLE_BRANCH/source.tar.gz
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com mkdir -p /workspace/shopsys/$CIRCLE_BRANCH/db
+          ssh -i ~/shopsys www-data@heimdall.shopsys.com chown -R www-data:www-data /workspace/shopsys/$CIRCLE_BRANCH
```

Comment:
The order of these commands make little sense to me. I would prefer:
- creating the workspace, changing its owner (`mkdir -p /workspace/shopsys/$CIRCLE_BRANCH && sudo chown -R www-data:www-data /workspace/shopsys/$CIRCLE_BRANCH`)
- cleaning the entire workspace on the cluster (`rm -rf /workspace/shopsys/$CIRCLE_BRANCH/*`)
- copying and extracting the sources (`scp ... && tar ... & rm source.tar.gz`)
- creating a directory for `db`

And that should be it. Having this part clean and understandable will be great for future changes.